### PR TITLE
Remove OSTATES from StatesTrajectory doxygen

### PR DESCRIPTION
**Pull request by chrisdembia**
This PR updates the StatesTrajectory documentation to no longer describe the OSTATES file format, as this is currently off the list for 4.0. We can easily add this documentation back as soon as OSTATES file support is in place.

This PR also addresses #1381 .

@tkuchida, would you be interested in reviewing this?

### CODE REVIEW

File: OpenSim/Simulation/StatesTrajectory.h [hunk 1]
```diff
@@ -44,20 +44,22 @@ class Model;
...
+ * StatesTrajectory during a simulation via the StatesTrajectoryReporter. You
+ * can also create a StatesTrajectory from a states storage (.sto) file (see
+ * createFromStatesStorage()). Users can modify a trajectory by appending
+ * states to it, but users cannot modify the individual states that are already
+ * in a trajectory.
  *
  * This class was introduced in OpenSim version 4.0, and enables scripting
  * (Python/MATLAB) and C++ users to postprocess their results with greater ease
- * and more versatility than with an Analysis.
+ * and more flexibly than with an Analysis.
```
Comment (tkuchida) [inline OpenSim/Simulation/StatesTrajectory.h]: "with greater ease and flexibility"?

File: OpenSim/Simulation/StatesTrajectory.h [hunk 2]
```diff
@@ -44,20 +44,22 @@ class Model;
...
  *
- * @note This class is a work in progress. The ability to read and write a
- * StatesTrajectory to an OSTATES is not available yet, even though the
- * documentation is written as if this ability exists.
+ * @note In a future release, we hope to support an OSTATES file format that
```
Comment (tkuchida) [inline OpenSim/Simulation/StatesTrajectory.h]: "hope" -> "plan"

**Comment by chrisdembia** [comment]
@tkuchida I have addressed your comments. 



> Built doxygen locally.

Ah! I could have omitted `[ci skip]` for the initial PR/commit. Sorry about that.

File: OpenSim/Simulation/StatesTrajectory.h [hunk 3]
```diff
@@ -44,20 +44,22 @@ class Model;
...
+ * StatesTrajectory during a simulation via the StatesTrajectoryReporter. You
+ * can also create a StatesTrajectory from a states storage (.sto) file (see
+ * createFromStatesStorage()). Users can modify a trajectory by appending
+ * states to it, but users cannot modify the individual states that are already
+ * in a trajectory.
  *
  * This class was introduced in OpenSim version 4.0, and enables scripting
  * (Python/MATLAB) and C++ users to postprocess their results with greater ease
- * and more versatility than with an Analysis.
+ * and flexibly than with an Analysis.
```
Comment (tkuchida) [inline OpenSim/Simulation/StatesTrajectory.h]: I suggested "flexibility" (with greater ease and [greater] flexibility than…); I don't think "flexibly" makes sense (with… flexibly than…).

Comment (chrisdembia) [inline OpenSim/Simulation/StatesTrajectory.h]: Oops; I didn't read carefully. I have resolved this now.

**Comment by tkuchida** [comment]
Looks good. Merging by [Rule 5](https://github.com/opensim-org/opensim-core/blob/master/CONTRIBUTING.md#making-a-pull-request-pr).

**Comment by chrisdembia** [comment]
Thank you.

**Pull request merged**